From: =?utf-8?q?R=C3=A9mi_Palancher?= <remi-externe.palancher@edf.fr>
Date: Mon, 29 Feb 2016 16:27:02 +0100
Subject: Backport to Cython 0.15.1

Report tested on Python 2.6.6, Cython 0.15.1 and Slurm 15.08.8 in
README.md.

Forwarded: https://github.com/gingergeeks/pyslurm/pull/67
---
 README.rst          | 1 +
 pyslurm/pyslurm.pyx | 4 +++-
 pyslurm/slurm.pxd   | 6 +++---
 3 files changed, 7 insertions(+), 4 deletions(-)

diff --git a/README.rst b/README.rst
index e05d288..4ecf6f1 100644
--- a/README.rst
+++ b/README.rst
@@ -23,6 +23,7 @@ This release has been tested with :
 	* Slurm 15.08.2,15.08.3, Cython 0.22.1 and Python 2.7.4
 	* Slurm 15.08.2,15.08.3 Cython 0.23.1 and Python 2.7.4
 	* Slurm 15.08.2,15.08.3 Cython 0.23.4 and Python 2.7.4
+	* Slurm 15.08.8 Cython 0.15.1 and Python 2.6.6
 
 Installation
 ************
diff --git a/pyslurm/pyslurm.pyx b/pyslurm/pyslurm.pyx
index a518afa..cdf4cc8 100644
--- a/pyslurm/pyslurm.pyx
+++ b/pyslurm/pyslurm.pyx
@@ -2415,7 +2415,9 @@ cdef class node:
 
 			# Enhanced node state - src/api/node_info.c
 
-			comp_str = drain_str = power_str = ''
+			comp_str = ""
+			drain_str = ""
+			power_str = ""
 			node_state = self._record.node_state
 			my_state = node_state
 
diff --git a/pyslurm/slurm.pxd b/pyslurm/slurm.pxd
index 5626a4f..92dfb0f 100644
--- a/pyslurm/slurm.pxd
+++ b/pyslurm/slurm.pxd
@@ -1721,8 +1721,8 @@ cdef extern from 'slurm/slurm.h' nogil:
 
 	cdef extern int slurm_job_cpus_allocated_on_node_id (job_resources_t *, int)
 	cdef extern int slurm_job_cpus_allocated_on_node (job_resources_t *, char *)
-	cdef extern int slurm_job_cpus_allocated_str_on_node_id (char *cpus, size_t cpus_len, job_resources_t *job_resrcs_ptr, int node_id)
-	cdef extern int slurm_job_cpus_allocated_str_on_node (char *cpus, size_t cpus_len, job_resources_t *job_resrcs_ptr, const char *node_name)
+	cdef extern int slurm_job_cpus_allocated_str_on_node_id (char *, size_t, job_resources_t *, int)
+	cdef extern int slurm_job_cpus_allocated_str_on_node (char *, size_t, job_resources_t *, const_char_ptr)
 
 	#
 	# Job Control Config
@@ -1895,7 +1895,7 @@ cdef extern from 'slurm/slurm.h' nogil:
 	# Burst Buffer
 	#
 
-	cdef extern char *slurm_burst_buffer_state_string (uint16_t state);
+	cdef extern char *slurm_burst_buffer_state_string (uint16_t state)
 	cdef extern int slurm_load_burst_buffer_info (burst_buffer_info_msg_t **burst_buffer_info_msg_pptr)
 	cdef extern void slurm_free_burst_buffer_info_msg (burst_buffer_info_msg_t *burst_buffer_info_msg)
 	cdef extern void slurm_print_burst_buffer_info_msg (FILE *out, burst_buffer_info_msg_t *info_ptr, int one_liner, int verbosity)
